server/src/main/java/org/elasticsearch/plugins/SearchPlugin.java

```diff
@@ -300,6 +303,23 @@ public AggregationSpec addResultReader(String writeableName, Writeable.Reader<?
         public Map<String, Writeable.Reader<? extends InternalAggregation>> getResultReaders() {
             return resultReaders;
         }
+
+        /**
+         * Get the function to register the {@link org.elasticsearch.search.aggregations.support.ValuesSource} to aggregator mappings for
+         * this aggregation
+         */
+        public Consumer<ValuesSourceRegistry> getRegisterAggregators() {
```

Comment:
Name seems a bit funky to me.  Perhaps something like `getAggregatorRegistrar()?